static/app/views/dashboardsV2/utils.tsx

```diff
@@ -13,16 +13,23 @@ export function cloneDashboard(dashboard: DashboardDetails): DashboardDetails {
 export function eventViewFromWidget(
   title: string,
   query: WidgetQuery,
-  selection: GlobalSelection
+  selection: GlobalSelection,
+  widgetType?: DisplayType
 ): EventView {
   const {start, end, period: statsPeriod} = selection.datetime;
   const {projects, environments} = selection;
 
+  // World Map requires an additional column (geo.country_code) to display in discover when navigating from the widget
+  const fields =
+    widgetType === DisplayType.WORLD_MAP
+      ? ['geo.country_code', ...query.fields]
+      : query.fields;
```

Comment:
If we are adding the `'geo.country_code'` field, should we also be adding `has:geo.country_code` to the query since I see we do that in the `/events-geo/` [endpoint](https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_events.py#L146)?

Reply:
👍  Added tests around widgetCard and Open in Discover links!
Also good point on `has:geo.country_code`, I've forced appending that to the end of the query when clicking Open in Discover. I'm not sure if we have a better utility for appending to queries, let me know if you're aware.

For your question about adding more options to the dropdown, seems like Dora already has plans to add additional options in the Figma. I'll ping you the link!